apps/src/applab/firebaseStorage.js

```diff
@@ -0,0 +1,260 @@
+'use strict';
+
+/* global Applab */
+
+var FirebaseUtils = require('./firebaseUtils');
+
+/**
+ * Namespace for Firebase storage.
+ */
+var FirebaseStorage = module.exports;
+
+function getKeysRef(channelId) {
+  var kv = FirebaseUtils.getDatabase(channelId).child('storage').child('keys');
+  return kv;
+}
+
+function getRecordsRef(channelId, tableName) {
+  return FirebaseUtils.getDatabase(channelId).child('storage').child('tables')
+    .child(tableName).child('records');
+}
+
+/**
+ * @param {string} tableName
+ * @returns {Promise<number>} next record id to assign.
+ */
+function getNextIdPromise(tableName) {
+  var lastIdRef = FirebaseUtils.getDatabase(Applab.channelId).child('counters').child('tables')
+      .child(tableName).child('last_id');
+  return lastIdRef.transaction(function (currentValue) {
+    return (currentValue || 0) + 1;
+  }).then(function (transactionData) {
+    return transactionData.snapshot.val();
+  });
```

Comment:
Done, here and throughout the file. Let me know if I went overboard and/or missed any opportunities for arrow functions